src/melee.cpp

```diff
@@ -1752,7 +1752,7 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
         ret.push_back(tmp);
     }
 
-    if (has_trait("INCISORS") && one_in(18 - dex_cur - get_skill_level( skill_unarmed )) &&
+    if (has_trait("INCISORS") && one_in(18 - dex_cur - get_skill_level( skill_unarmed )) && (elem.has_flag( "ALLOWS_NATURAL_ATTACKS" )) &&
```

Comment:
You should wrap `is_wearing_something_on` and `has_flag( "ALLOWS"` into one check:
`exclusive_flag_coverage( "ALLOWS_NATURAL_ATTACKS" )`
Since `exclusive_flag_coverage` returns a bitset, you could use it like that:
```
const auto uncovered_part = exclusive_flag_coverage( "ALLOWS_NATURAL_ATTACKS" );
if( has_trait( "HORNS" ) && uncovered_part[bp_head] ) {...
}
if( has_trait( "CLAW_FOOT" ) && uncovered_part[bp_leg] ) {
...
```

Or better - wrap it in a function.

Your current check is wrong, because fingerless gloves + regular gloves shouldn't allow scratching, while in your code they do. Using `exclusive_flag_coverage` would make it check if all items allow attacks, not if any of them does.